clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -186,6 +188,101 @@ public void testMetricsReporterAutoGeneratedClientId() {
         consumer.close();
     }
 
+    @Test
+    public void testPollReturnsRecords() {
+        KafkaConsumer<String, String> consumer = setUpConsumerWithRecordsToPoll(tp0, 5);
+
+        ConsumerRecords<String, String> records = consumer.poll(Duration.ZERO);
+
+        assertEquals(records.count(), 5);
+        assertEquals(records.partitions(), new HashSet<>(Collections.singletonList(tp0)));
+        assertEquals(records.records(tp0).size(), 5);
+
+        consumer.close(Duration.ofMillis(0));
+    }
+
+    @Test
+    public void testPollReturnsRecordsUpUntilError() {
```

Comment:
As far as I can tell, the logic in this test case is the same as in the first half of `testSecondPollWithDeserializationErrorThrowsRecordDeserializationException`. Do we need both of them?

Reply:
Should I merge into one method? One is checking you get successful records UP to the poison pill. 

The other test is the throwing of the exception

Reply:
combined into one.

Reply:
done